extras/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisReceiver.scala

```diff
@@ -65,11 +69,14 @@ import com.amazonaws.services.kinesis.clientlibrary.lib.worker.Worker
 private[kinesis] class KinesisReceiver(
     appName: String,
     streamName: String,
-    endpointUrl: String,
+    endpointUrl: String, 
+    regionName: String,
+    awsAccessKeyId: String,
```

Comment:
If these can be null, then better to make these use `Option[AWSCredentials]` where `case class AWSCredentials(accessKeyId: String, secretKey)`. And place it at the end of the parameter list.